### DISCUSSION TITLE
Nvidia Broadcast for Linux

### DISCUSSION THREAD

**Discussion by edisionnano** [NVIDIA/open-gpu-kernel-modules#305] [Ideas]
Nvidia broadcast also known as RTX voice can remove audio and video background noise on NVIDIA GPUs. This is only available for Windows but the Video and Audio AI SDKs it uses are available for Linux too but not yet public. Linux gamers would really benefit from at least RTX voice since krisp isn't available for Linux (unless Android and the discontinued Chrome plugin count) and Noisetorch was recently compromised. Nvidia, you can give us this. Even if it doesn't have a GUI it will be still fine, the community can make one.

**Comment by AlexvZyl** [comment]
This would really be awesome.  One of the last things that Windows has over Linux.

**Reply by vitorklock** [reply]
For me, it's THE last thing holding me back.

**Comment by mattmattox** [comment]
Do we know if this is on the road map?

**Comment by rodrigoGA** [comment]
It is incredible how NVIDIA despises Linux users. Nvidia strives to restrict the use of its technologies in linux and to provide poor quality products, even from the drivers.
I think that linux users should take this into account before choosing nvidia, because after all we can't use it, and alternatives provide solutions that may not be at the same technological level, but work on linux.
It's a pity that such an innovative company forces users to choose other alternatives due to commercial decisions.

**Reply by AlexvZyl** [reply]
I wish I could upvote twice.

**Reply by a-y-u-s-h** [reply]
I know you copied this from a comment on Reddit but okay, I will upvote this anyway.

**Reply by erzwo** [reply]
That's the reason, my next gpu will be team red.

**Reply by turowicz** [reply]
NVIDIA is very much on top of Linux for workstation / server workflows.

**Reply by vitorklock** [reply]
If this doesn't come out, I'm putting my green away into a red one.

**Comment by twidi** [comment]
Seems it's currently in development https://developer.nvidia.com/broadcast-engine/video-effects-sdk-linux

**Reply by AlexvZyl** [reply]
Exciting.

**Reply by edisionnano** [reply]
That has been a thing for quite a while. It's just the sdk for devs...

**Reply by johnycage** [reply]
It's currently "invite only program"

**Reply by RiemaruKarurosu** [reply]
I applied months ago, still pending to review my request to access.

**Comment by thefonso** [comment]
2023...no forward motion...I guess we know where this is going

**Comment by kulievvitaly** [comment]
That is good product. I want to use it with ubuntu.

**Reply by writetoanikets** [reply]
I tried Cadmus but  I don't like it

**Reply by JanisPlayer** [reply]
You can try this, I was surprised, it even filters a vacuum cleaner. Perfect for my jet engine hot air gun notebook.
https://github.com/Rikorose/DeepFilterNet

**Comment by iamkhalidbashir** [comment]
Shame!

**Comment by orowith2os** [comment]
Krisp should be available for Discord on Linux now, and you can use EasyEffects to filter out background noise in other apps.

**Reply by RiemaruKarurosu** [reply]
what about removing camera background?

**Comment by j2l** [comment]
I upvoted here, and while continuing the quest, stumbled on this: https://catalog.ngc.nvidia.com/orgs/nvidia/teams/maxine/resources/maxine_linux_video_effects_sdk_ga
It's only for video effects.
Looks like it's 3 months old (April 2023) but the requirement (in the README file) states it's for ubuntu 18.04 and 20.04, so "only" 3-5 years late :smile: 

I switched to pop 22.04 so I can't test (another Ubuntu EEE strategy  based on `uname -r` (Embrace, Extend, Extinguish other Ubuntus, far from Madiba's legacy)), but if anyone here has a "official" 20.04 and can confirm it works, that could be beginning of some coolness.

**Reply by knileshh** [reply]
Does, pop have this functionality?

**Reply by j2l** [reply]
Why shouldn't it have it? It's only ubuntu + Nvidia proprietary drivers

**Reply by RiemaruKarurosu** [reply]
Could someone even could download it?, it gives me back 451 Unavailable For Legal Reasons

**Comment by gelodefaultbrain** [comment]
Any news guys? is it available now?

**Comment by antonsem** [comment]
I'm thinking about permanently switching to Linux. This is the only app I'm actively using which doesn't have a proper alternative. I would love to see it ported...

**Comment by Inmoresentum** [comment]
Please give us access to the SDK.... We can then take it from there.

**Reply by RiemaruKarurosu** [reply]
I don't know why they don't want to give us access.

**Comment by gelodefaultbrain** [comment]
> Any news guys? is it available now?


Replying to my comment here. It's now almost 2024, is there any news? I really wanted to have it on my Ubuntu machine tho so that it can reduce the background noise of my dog barking.

**Comment by orowith2os** [comment]
Saying it again, instead of spamming this thread with questions and questions that very clearly will never result in any answers, just use EasyEffects, and it appears recent versions of Discord support Krisp as well.

**Reply by RiemaruKarurosu** [reply]
EsayEffects does not include nvidia video AI modifications (and there are not any good alternative)

**Comment by myagizmaktav** [comment]
Any Update?

**Comment by JayPokale** [comment]
Windows users to Linux users: I feel sad for you

Linux users to Windows users: I don't think about you at all

**Comment by chriskalos** [comment]
I want to switch to Linux full time after the horrible business practices of Microsoft, but I just cannot let go of Nvidia Broadcast. Nvidia NEEDS to prioritize this.

**Comment by gelodefaultbrain** [comment]
Guys? Any updates? Had my last comment [here](https://github.com/NVIDIA/open-gpu-kernel-modules/discussions/305#discussioncomment-7689176). It's almost 2025 again.

**Comment by mtijanic** [comment]
Just to set expectations on this thread: This is the _kernel modules_ repo, that is monitored by developers working on the kernel modules. The feature in question has nothing to do with the kernel modules, and none of us here can give you any info that you are asking for.

I'm not sure where would be a good place to reach the people who could have some insight into this, but it definitely isn't here. Best bet is the general linux forum: https://forums.developer.nvidia.com/c/gpu-graphics/linux/148

Feel free to keep this thread going (as long as its civil) if you want to discuss stuff, but do not expect any sort of official answer here whatsoever.

**Reply by darkhog** [reply]
Yeah but it would be nice if you could send this to the appropriate department.

**Comment by darkhog** [comment]
My main use for Broadcast is removing background from webcam (can't afford a green screen). Is there any alternative on Linux for THAT?

**Comment by roworu** [comment]
There is a dedicated page on the Nvidia forums about the Broadcast app on Linux. If you are interested, you can go and upvote it: https://www.nvidia.com/en-us/geforce/forums/broadcasting/18/483398/nvidia-broadcast-app-for-linux/
I am not sure if bumping a GitHub issue in the GPU kernel modules repository would speed up the process.

**Comment by zerofill** [comment]
Amazing that 90% of the AI use of Nvidia cards is on Linux, but they don't care to make their apps work on Linux.

**Reply by arisu3** [reply]
Nvidia has very good support for HPC and ML applications on Linux when using their datacenter-class cards. Everything else, Windows or Linux, is almost an afterthought.

**Comment by Aloosh101** [comment]
Hi, now it's 2025 and I'm waiting impatiently for this technology even though I use nixos but I have no problem trying to contribute to make it compatible with this system I'm just waiting for any updates

**Comment by angelbruni** [comment]
This is currently the only thing I need from NVIDIA on Linux.

**Reply by darkhog** [reply]
Agreed, the only alternative is some shitty OBS plugin that freezes when trying to use hardware acceleration on it.

**Comment by delorenj** [comment]
pleeeeeeeease make this

**Reply by myagizmaktav** [reply]
Im waiting this feature since a year...

**Comment by dlukt** [comment]
Windows 10 deprecation is coming and with it many people who will not "upgrade" to Windows 11 and will look for alternatives.
Gaming on Linux is in a good place right now

**Comment by thelittlebug** [comment]
![image](https://github.com/user-attachments/assets/22689a4f-a068-4382-85ed-c8ab32062fb2)

**Reply by myagizmaktav** [reply]
You send it not full version here is <img src="https://cdn.discordapp.com/avatars/145989184744587266/a_dd973908e6e46ce3308f6b1a36da3b18.gif?size=1024" size="500px" />

**Comment by orcinmansam** [comment]
I would love to see this make its way to linux. I don't care for the video stuff, just the sound scrubbing is absolutely game changing

**Reply by JanisPlayer** [reply]
If you only care about the sound, use something similar like https://github.com/Rikorose/DeepFilterNet. Nvidia generally only does the bare minimum for Linux. You can be happy if your driver doesn't randomly freeze and encoders work. I don't think they'll recreate the proprietary program for Linux anytime soon.

**Reply by darkhog** [reply]
Regarding sound, I've achieved good noise reduction with EasyEffects software (requires PipeWire AFAIK though), but for video there's pretty much nothing. I've tried to use that background removal OBS plugin, but it is crap compared to Broadcast.

**Reply by Bo0sted** [reply]
Agreed yea. NVIDIA Broadcast is miles ahead of that OBS plugin

**Comment by mexicantexan** [comment]
Would love to see this ported to linux!

**Reply by mexicantexan** [reply]
Seems like it should be easy enough given the tooling that already exists for linux users, but what do I know.

**Reply by darkhog** [reply]
You can always try doing it. I'm sure many people would appreciate it.

**Reply by mexicantexan** [reply]
Want to help?

**Reply by pkp24** [reply]
> Want to help?

you said it should be easy enough. get to it

**Reply by mexicantexan** [reply]
Don't have time at the moment unfortunately

**Reply by mexicantexan** [reply]
MVP for the Linux based Broadcast app that relies on the Maxine SDK: https://github.com/mexicantexan/studiocast
Right now it replies on the NVIDIA Maxine SDK, but I'll incorporate open sourced near-equivalents within the next week. Tested on Ubuntu 22.04 & Ubuntu 24.04 with a RTX 2070.

**Comment by turowicz** [comment]
Probably could work as a cross platform OSS project, that NVIDIA will never do. Nothing stops the community from building it though.

**Reply by darkhog** [reply]
Well, things are changing. GeForce NOW is now on Linux, maybe Broadcast will be next? All they really need to do is to create a virtual V4L or V4L2 device to send processed image data to that can then be accessed by OBS.

**Comment by aarmn** [comment]
**open**-gpu-kernel-modules like **Open**AI

**Comment by mexicantexan** [comment]
MVP for the Linux based Broadcast app that relies on the Maxine SDK: https://github.com/mexicantexan/studiocast
Right now, it relies on the NVIDIA Maxine SDK, but I'll incorporate open-sourced near-equivalents within the next week. Tested on Ubuntu 22.04 & Ubuntu 24.04 with a RTX 2070.

Feedback welcome, lol again MVP 

edit: note - the repo has no affiliation with NVIDIA, I'm just wanting to get off Windows ASAFP and this has been one of the things holding me back.

**Comment by Pedrojok01** [comment]
Same here, Nvidia Broadcast is basically the only app I'm missing on Linux!
If anyone is interested, I've created his: [https://github.com/Pedrojok01/linux-broadcast](https://github.com/Pedrojok01/linux-broadcast).
No CUDA, no PyTorch, no Python. Just a quick and simple install (.deb and Arch so far). Already using it in all my meetings.  
It is only the video part though, no audio. 
Contributions and feedback are welcome.

**Comment by mexicantexan** [comment]
Hey Pedrojok01 I like what you did with the UI! Mind if I incorporate it
into mine?

Respectfully,
Jacob Dallas

**Reply by Pedrojok01** [reply]
Hi @mexicantexan , 
Thanks for the nice feedback. I  don't mind, go ahead :+1:
